public/docs/_examples/server-communication/e2e-spec.js

```diff
@@ -0,0 +1,174 @@
+describe('Server Communication', function () {
+
+  beforeAll(function () {
+    browser.get('');
+  });
+
+  describe('Tour of Heroes e2e tests', function () {
+    
+    var _initialHeroCount = 4;
+    var _newHeroName = 'Mr. IQ';
+    var _heroCountAfterAdd = 5;
+    
+    it('should display ' + _initialHeroCount + 'heroes after init', function () {
+      var myTohComp = element(by.tagName('my-toh'));
+      expect(myTohComp).toBeDefined('<my-toh> must exist');
+      var heroListComp = myTohComp.element(by.tagName('hero-list'));
+      expect(heroListComp).toBeDefined('<hero-list> must exist');
+      var heroTags = heroListComp.all(by.tagName('li'));
+      expect(heroTags.count()).toBe(_initialHeroCount);
+    });
+    
+    it('should not add hero with empty name', function () {
+      var myTohComp = element(by.tagName('my-toh'));
+      expect(myTohComp).toBeDefined('<my-toh> must exist');
+      var addButton = myTohComp.element(by.tagName('button'));
+      expect(addButton).toBeDefined('"Add Hero" button must be defined');
+      addButton.click(function() {
```

Comment:
The following kind of test, which you use repeatedly, does not work 
```
addButton.click(function() {...})
```
The function is never executed; the test always passes.

The `click` method returns a promise. The correct syntax invokes the function within the `then` success callback:
```
addButton.click().then(function() { ... })
```